tests/src/rules/no-cycle.js

```diff
@@ -290,4 +290,25 @@ ruleTester.run('no-cycle', rule, {
       ],
     }),
   ),
+};
+
+ruleTester.run('no-cycle', rule, {
+  valid: cases.valid.map((testCase) => [testCase, {
+    ...testCase,
+    code: `${testCase.code} // skipErrorMessagePath=true`,
+    options: [{
+      ...testCase.options && testCase.options[0] || {},
+      skipErrorMessagePath: true,
+    }],
+  }]).flat(1),
```

Comment:
we can't rely on `.flat` existing, but i think you csn do this with `[].concat()` instead

Reply:
In production code, sure - but this is test code.
I can replace it with `[].concat()` manually but I believe it's better to use a polyfill for this.
If we don't go with polyfill then we ought to add an ESLint rule to error when using `Array.flat`.

Reply:
installing a polyfill would invalidate the tests, so that's not a good idea.
we can import `array.prototype.flat`, certainly, but `[].concat` is pretty idiomatic - either way is fine.

it's not possible to add an eslint rule that can reliably warn against instance method usage without type information.

Reply:
> it's not possible to add an eslint rule that can reliably warn against instance method usage without type information

False-positives would be possible in theory, you're correct that it won't be reliable. But how bad would it be practically speaking?

Reply:
It'd add complication without much value; the tests will already fail as-is due to the function being unavailable.

Reply:
replaced `array.flat()` with `[].concat(...array)`

- https://github.com/import-js/eslint-plugin-import/pull/2998/commits/a6b26c461e6f14b40257aaa769128aeb3203c9f3